core/network/impl/reputation_repository_impl.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "network/impl/reputation_repository_impl.hpp"
+
+#include <boost/assert.hpp>
+
+using namespace std::chrono_literals;
+
+namespace kagome::network {
+
+  ReputationRepositoryImpl::ReputationRepositoryImpl(
+      std::shared_ptr<libp2p::basic::Scheduler> scheduler)
+      : scheduler_{std::move(scheduler)} {
+    BOOST_ASSERT(scheduler_);
+
+    tick_handler_ = scheduler_->scheduleWithHandle([&] { tick(); }, 1s);
+  }
+
+  Reputation ReputationRepositoryImpl::reputation(
+      const ReputationRepository::PeerId &peer_id) const {
+    return rating_table_.count(peer_id) == 0 ? 0 : rating_table_.at(peer_id);
+  }
+
+  Reputation ReputationRepositoryImpl::change(
+      const ReputationRepository::PeerId &peer_id, ReputationChange diff) {
+    ensurePeerPresence(peer_id);
+    return rating_table_[peer_id] += diff.value;
+  }
+
+  void ReputationRepositoryImpl::ensurePeerPresence(const PeerId &peer_id) {
+    rating_table_.try_emplace(peer_id, 0);
+  }
```

Comment:
```suggestion
  Reputation ReputationRepositoryImpl::reputation(
      const ReputationRepository::PeerId &peer_id) const {
    const auto it = rating_table_.find(peer_id);
    return it == rating_table_.end() ? 0 : it->second;
  }

  Reputation ReputationRepositoryImpl::change(
      const ReputationRepository::PeerId &peer_id, ReputationChange diff) {
    return rating_table_[peer_id] += diff.value;
  }
```

Reply:
This peace of code was refactored (rename of class).
Probably @igor-egorov can explain reason of that implementation.

Reply:
Sure, the code could be refactored. Both options are perfectly fine